### DISCUSSION TITLE
有没有可能“劫持”chrome浏览器自带的“翻成中文”右键菜单？

### DISCUSSION THREAD

**Discussion by JoeyLearnsToCode** [immersive-translate/immersive-translate#533] [Ideas]
### 新功能 | New Feature

![image](https://github.com/immersive-translate/immersive-translate/assets/11903658/695a6294-dd5b-4b9c-b7e4-f301518e5054)

如上图，chrome浏览器右键菜单中自带了 “翻成中文” 的翻译功能，技术上有没有可能在“劫持”这个浏览器行为，调用沉浸翻译的网页翻译。
我通过 TemperMonkey 使用沉浸翻译脚本。

最后，向你们表示感谢，我今天刚发现这个脚本，真是一个让人感动的好产品。

### 为什么您想要添加这些功能 | Why you want to add these features

能够让使用体验更自然，就好像chrome自带的翻译功能就天然支持了双语对照翻译一样。

### 提议 | Proposal

浏览器可能会为自己的行为提供 hook api 或者hijack api，如果有，脚本、拓展应该可以利用之。

**Comment by versun** [comment]
"不可能, 绝不可能"😄

**Comment by reece00** [comment]
谷歌：6，哪来的大聪明，专门做劫持自己菜单的功能

**Reply by JoeyLearnsToCode** [reply]
我说的劫持不是那种恶意的劫持，相当于拦截器模式，允许默认的行为被修改
